api/v1alpha1/api_key_auth_types.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+const APIKeysSecretKey = "credentials"
+
+// APIKeyAuth defines the configuration for the API Key Authentication.
+type APIKeyAuth struct {
+	// Credentials is the Kubernetes secret which contains the API keys.
+	// This is an Opaque secret.
```

Comment:
we should highlight the format of the `Secret` here , its a map of key - value pairs, is the map key the `API Key` and the map value the `Client ID` or vice versa

Reply:
Isn't the existing doc enough? (or is this explanation of mine unclear?

```go
 	// Each API key is stored in the key representing the client id,
	// which can be used in AllowedClients to authorize the client in a simple way.
```

Reply:
sure, lets keep this for now, and can improve in the future